compiler/parser/RdrHsSyn.hs

```diff
@@ -2533,32 +2533,35 @@ data TemplateBodyDecl
   | KeyDecl (Located KeyData)
   | MaintainerDecl (LHsExpr GhcPs)
 
+data TemplateBodyDecls =
+  TemplateBodyDecls {
+      tbdEnsures :: [LHsExpr GhcPs]
+    , tbdSignatories :: [LHsExpr GhcPs] -- (list of lists)
+    , tbdObservers :: [LHsExpr GhcPs] -- (list of lists)
```

Comment:
I'd be tempted to comment them as `each HsExpr has type `[Party]` or similar, and ditto for all the others, e.g. tbdEnsures.

Reply:
It gets too hard. Controlled choice groups aren't so easily described 😄  The list of lists things was a reminder to me once upon a time to `applyConcat`. I've deleted the comments entirely in the next revision.